pontoon/administration/management/commands/sync_projects.py

```diff
@@ -0,0 +1,363 @@
+            else:
+                # VCS no longer has the entity, remove it from Pontoon.
+                changeset.obsolete_db_entity(db_entity)
+        elif db_entity is None:
+            # New VCS entities are added to Pontoon.
+            changeset.create_db_entity(vcs_entity)
+        else:
+            for locale in db_project.locales.all():
+                if db_entity.has_changed(locale.code):
+                    # Pontoon changes overwrite whatever VCS has.
+                    changeset.update_vcs_entity(locale.code, db_entity, vcs_entity)
+                elif vcs_entity.has_translation_for(locale.code):
+                    # If Pontoon has nothing or has not changed, and the VCS
+                    # still has the entity, update Pontoon with whatever may
+                    # have changed.
+                    changeset.update_db_entity(locale.code, db_entity, vcs_entity)
+
+    def update_resources(self, db_project, vcs_project):
+        """Update the database on what resource files exist in VCS."""
+        relative_paths = vcs_project.resources.keys()
+        db_project.resource_set.exclude(path__in=relative_paths).delete()
+
+        for relative_path, vcs_resource in vcs_project.resources.items():
+            resource, created = db_project.resource_set.get_or_create(path=relative_path)
+            resource.format = Resource.get_path_format(relative_path)
+            resource.entity_count = len(vcs_resource.entities)
+            resource.save()
+
+    def get_vcs_entities(self, vcs_project):
+        return {self.entity_key(entity): entity for entity in vcs_project.entities}
+
+    def get_db_entities(self, db_project):
+        entities = (Entity.objects
+                    .select_related('resource')
+                    .filter(resource__project=db_project, obsolete=False))
+        return {self.entity_key(entity): entity for entity in entities}
+
+    def entity_key(self, entity):
+        """
+        Generate a key for the given entity that is unique within the
+        project.
+        """
+        return ':'.join([entity.resource.path, entity.string])
+
+    def commit_changes(self, db_project):
+        """Commit the changes we've made back to the VCS."""
+        # TODO: Set translation author properly.
+        user = User.objects.filter(is_superuser=True).first()
+
+        for locale in db_project.locales.all():
+            commit_message = (
+                'Pontoon: Updated {locale.name} ({locale.code}) localization of '
+                '{project.name}'
+            ).format(locale=locale, project=db_project)
+
+            try:
+                result = commit_to_vcs(
+                    db_project.repository_type,
+                    db_project.locale_directory_path(locale.code),
+                    commit_message,
+                    user,
+                    db_project.repository_url
+                )
+            except Exception as err:
```

Comment:
This gives me flashbacks to overly zealous exception handlers that disguised typos in plugins and made things a lot harder to debug. Is it possible to narrow the caught types of exceptions?